share/trick/sim_objects/default_trick_sys.sm

```diff
@@ -175,7 +175,7 @@ class MonteCarloSimObject : public Trick::SimObject {
             exec_register_scheduler(&mc) ;
 
             {TRK} P0 ("default_data")   mc.process_sim_args() ;
-            {TRK} P0 ("initialization") mc.execute_monte() ;
+            {TRK} P1 ("initialization") mc.execute_monte() ;
```

Comment:
FYI @jdeans289 @jmpenn I had to increase the phase by one here `P1` so that the new method `MonteCarloGenerate.sm:41` executes first at `P0`. Without this change, a user mis-configuring a SIM for both old and new MC never hits the error checking in `MonteCarloGenerate.sm:43` if `trick_mc.mc.set_dry_run(True)`. Let me know if you have objection to this or another way to get the same behavior.